hardhat.config.ts

```diff
@@ -0,0 +1,140 @@
+import "@nomiclabs/hardhat-waffle";
+import "@nomiclabs/hardhat-ethers";
+import "@tenderly/hardhat-tenderly";
+import "@nomiclabs/hardhat-etherscan";
+import "@nomiclabs/hardhat-web3";
+import "hardhat-deploy";
+import "hardhat-deploy-ethers";
+import "@typechain/hardhat";
+
+import { resolve } from "path";
+import { config as dotenvConfig } from "dotenv";
+import { HardhatUserConfig } from "hardhat/config";
+import { NetworkUserConfig } from "hardhat/types";
+import { utils } from "ethers";
+import Web3 from "web3";
+
+dotenvConfig({ path: resolve(__dirname, "./.env") });
+
+const chainIds = {
+  ganache: 1337,
+  goerli: 5,
+  hardhat: 31337,
+  kovan: 42,
+  mainnet: 1,
+  rinkeby: 4,
+  ropsten: 3,
+  avalanche: 43114,
+  polygon: 137,
+};
+
+// Ensure that we have all the environment variables we need.
+const mnemonic = process.env.MNEMONIC;
+if (!mnemonic) {
+  throw new Error("Please set your MNEMONIC in a .env file");
+}
+
+const alchemyApiKey = process.env.ALCHEMY_API_KEY;
+if (!alchemyApiKey) {
+  throw new Error("Please set your ALCHEMY_ETH_API_KEY in a .env file");
+}
+
+const PRIVATE_KEY = process.env.PRIVATE_KEY;
+const ETHERSCAN_API_KEY = process.env.ETHERSCAN_API_KEY;
+
+function createTestnetConfig(network: keyof typeof chainIds): NetworkUserConfig {
+  const url: string = "https://eth-" + network + ".alchemyapi.io/v2/" + alchemyApiKey;
+  return {
+    accounts: {
+      count: 10,
+      initialIndex: 0,
+      mnemonic,
+      path: "m/44'/60'/0'/0",
+    },
```

Comment:
If we remove the above `mnemonic` condition, this will change. Eg: If we add private key in .env, it will use private key otherwise it will check if we have `mnemonic` and use `mnemonic`

Reply:
why don't we leave it as it is?

Reply:
why don't we leave it as it is?

Reply:
Because if we don't pass `mnemonic` or keep `mnemonic` .env undefined. Hardhat will throw an error. And currently, we it doesn't support private key, right?

Reply:
But having a mnemonic is better than using the private key.
Private key is only used at time of deployment.

Reply:
But why would I want to pass a `mnemonic` for test cases? We can directly use default `mnemonic` by hardhat, right?

Reply:
Sure, we can!
Then we should remove it

Reply:
Don't remove it. Just if `process.env.MNEMONIC` is not there. Don't use it. And check for `privateKey` instead

Reply:
okay

Reply:
done and pushed